app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+class User < ActiveRecord::Base
+  has_many :recipes, dependent: :destroy
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true,
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: true
+  has_secure_password
+
+  before_create :generate_auth_token
+  after_create :generate_recipes
+
+private
+  def generate_auth_token
+    begin
+      generated_token = SecureRandom.hex(10)
+    end while User.where(auth_token: generated_token).exists?
+
+    self.auth_token = generated_token
+  end
+
+  def generate_recipes
+    if self.seed_recipes == true
+      recipes = YAML::load(File.open("#{Rails.root}/config/recipies.yml"))
+      recipes.each do |recipe|
+        self.recipes.create(
+          name: recipe["recipe"]["name"],
+          difficulty: recipe["recipe"]["difficulty"],
+          instructions: recipe["recipe"]["instructions"],
+          favorite: recipe["recipe"]["favorite"],
+          remote_photo_url: recipe["recipe"]["remote_photo_url"],
+          description: recipe["recipe"]["description"])
+      end
+    end
+  end
```

Comment:
> Is it right to host the new class in models directory? if not where should it go?

Good question! According to [Getting Started with Rails](http://guides.rubyonrails.org/getting_started.html), `lib` seems an appropriate place for this kind of thing. I agree with it. Lots of people put it in `app` or even `app/models`, though, so don't feel bad.

> Where should the recipes yml file be hosted? is it okay having it in the config directory?

I think either `config` or `db` works. I could go either way.